skopt/optimizer/optimizer.py

```
@@ -176,7 +183,109 @@ def _check_arguments(self, base_estimator, n_random_starts, acq_optimizer):
+
+            - If set to `"cl_min"`, then constant liar strtategy is used
+               with lie objective value being minimum of observed objective
+               values. `"cl_mean"` and `"cl_max"` means mean and max of values
+               respectively. For details on this strategy see:
+
+               https://hal.archives-ouvertes.fr/hal-00732512/document
+
+               With this strategy a copy of optimizer is created, which is
+               then asked for a point, and the point is told to the copy of
+               optimizer with some fake objective (lie), the next point is
+               asked from copy, it is also told to the copy with fake
+               objective and so on. The type of lie defines different
+               flavours of `cl_x` strategies.
+
+        """
+        if n_points is None:
+            return self._ask()
+
+        supported_strategies = ["cl_min", "cl_mean", "cl_max"]
+
+        if not (isinstance(n_points, int) and n_points > 0):
+            raise ValueError(
+                "n_points should be int > 0, got " + str(n_points)
+            )
+
+        if strategy not in supported_strategies:
+            raise ValueError(
+                "Expected parallel_strategy to be one of " +
+                str(supported_strategies) + ", " + "got %s" % strategy
+            )
+
+        # Caching the result. If some new parameters are provided to the ask,
+        # the cache is not used.
+        if (n_points, strategy) in self.cache:
+            return self.cache[(n_points, strategy)]
+
+        # Copy of the optimizer is made in order to manage the
+        # deletion of points with "lie" objective (the copy of
+        # oiptimizer is simply discarded)
+        opt = self.copy()
+
+        X = []
+        for i in range(n_points):
+            x = opt.ask()
+            X.append(x)
+            if strategy == "cl_min":
+                y_lie = np.min(opt.yi) if opt.yi else 0.0  # CL-min lie
```

Should we just raise an error when `n_random_starts` is lesser than `n_points`. Where else is `opt.yi` an empty list?